Assets/Scripts/Structure/DoorAnimator.cs

```diff
@@ -1,177 +1,118 @@
-        playerLayer = LayerMask.NameToLayer("Player");
     }
 
     public override void OnStartClient()
     {
         base.OnStartClient();
-        SyncInitialState();
-    }
-
-    void SyncInitialState()
-    {
-        if (isServer) return;
-        if (isOpened)
-        {
-            leftPanel.localPosition = openLeft;
-            rightPanel.localPosition = openRight;
-        }
+        
+        if (isClientOnly)
+            SetDoorState(openState);
     }
     
     public void OnTriggerEnter(Collider other)
     {
-        
-        if (other.gameObject.layer != playerLayer || !isServer) return;
-        
-        if (!insidePlayers.Contains(other.gameObject))
+        if ((other.gameObject.layer & doorTriggerLayers) == 0 || !isServer) return;
+
+        if (playersInTrigger == 0)
         {
-            insidePlayers.Add(other.gameObject);
+            if (closeTimer != null) StopCoroutine(closeTimer);
+            openState = true;
         }
+
+        playersInTrigger += 1;
     }
 
     public void OnTriggerExit(Collider other)
     {
-        if (other.gameObject.layer != playerLayer || !isServer) return;
-        
-        if (insidePlayers.Contains(other.gameObject))
+        if ((other.gameObject.layer & doorTriggerLayers) == 0 || !isServer) return;
+
+        if (playersInTrigger == 1)
         {
-            insidePlayers.Remove(other.gameObject);
+            // Start the close timer (which may be stopped).
+            closeTimer = StartCoroutine(RunCloseEventually());
         }
-    }
 
-    void OpenDoor()
-    {
-        if (isClosingLocal || isOpeningLocal) return;
-        
-        lerpTime = 0f;
-        StartCoroutine(OpenDoorAnim());
-        openSfx.Play();
+        playersInTrigger = Math.Max(playersInTrigger - 1, 0);
     }
 
-    void CloseDoor()
+    private void SetDoorState(bool open)
     {
-        if (isClosingLocal || isOpeningLocal) return;
-        
-        lerpTime = 0f;
-        StartCoroutine(CloseDoorAnim());
-        closeSfx.Play();
-    }
+        if (openState == open)
+            return;
+        openState = open;
 
-    IEnumerator OpenDoorAnim()
-    {
-        if(isServer) isOpening = true;
-        isOpeningLocal = true;
-        while (lerpTime < 1f)
+        if (isClient)
         {
-            lerpTime += Time.deltaTime * (openSpeed * 2f);
-
-            leftPanel.localPosition = Vector3.Lerp(closedLeft, openLeft, lerpTime);
-            rightPanel.localPosition = Vector3.Lerp(closedRight, openRight, lerpTime);
-            yield return new WaitForEndOfFrame();
+            (open ? openSfx : closeSfx).Play();
+            StartCoroutine(RunDoorAnim(open));
         }
-        
-        yield return new WaitForSeconds(1f);
-
-        if(isServer) isOpening = false;
-        isOpeningLocal = false;
-        if(isServer) isOpened = true;
     }
 
-    IEnumerator CloseDoorAnim()
+    /*
+     * Runs the animation.
+     * TODO: Replace this with a real animation.
+     */
```

Comment:
I don't recommend using MechAnim for this as many instances of MechAnim consume more cpu time and also affect the awake time of a scene on load. That is why I animated it using lerp methods

Reply:
I can see your point with that, however even if we do not change the animation, modifications need to be made - see double doors.

Reply:
Using mecanim offers us a huge advantage regarding dynamism in our animations.
client-wise the performance loss would be minimal thanks to having to manage only what is on screen.